kernel/arch/dreamcast/include/dc/dmac.h

```diff
@@ -0,0 +1,129 @@
+/** \file   dc/dmac.h
+    \brief  Macros to access the DMAC(Direct Memory Access Controller) 
+    registers. 
+
+    This header provides a set of macros to facilitate checking
+    the values of various DMA channels on the system.
+
+    DMA channel 0 and its registers (DMAC_SAR0, DMAC_DAR0, DMAC_DMATCR0, 
+    DMAC_CHCR0) are used by the hardware and not accessible to us but are 
+    documented here anyway.
+
+    DMA channel 2 is strictly used to transfer data to the PVR/TA.
+
+    DMA channel 1 & 3 are free to use.
+*/
+
+#ifndef __DC_DMAC_H
+#define __DC_DMAC_H
+
+#include <sys/cdefs.h>
+__BEGIN_DECLS
+
+#define DMAC_BASE     0xffa00000
+
+/** \defgroup dma_sar_registers  DMA Source Address Registers (SAR0-SAR3)
+  
+    These registers designate the source address for DMA transfers. 
+    Currently we only support 32-byte boundary addresses.
+
+    @{
+ */
+
+#define DMAC_SAR0     (*((vuint32 *)(DMAC_BASE + 0x00)))
+#define DMAC_SAR1     (*((vuint32 *)(DMAC_BASE + 0x10)))
+#define DMAC_SAR2     (*((vuint32 *)(DMAC_BASE + 0x20)))
+#define DMAC_SAR3     (*((vuint32 *)(DMAC_BASE + 0x30)))
+
+/** @} */
+
+/** \defgroup dma_dar_registers  DMA Destination Address Registers (DAR0-DAR3)
+
+    These registers designate the destination address for DMA transfers. 
+    Currently we only support 32-byte boundary addresses.
+
+    @{
+ */
+
+#define DMAC_DAR0     (*((vuint32 *)(DMAC_BASE + 0x04)))
+#define DMAC_DAR1     (*((vuint32 *)(DMAC_BASE + 0x14)))
+#define DMAC_DAR2     (*((vuint32 *)(DMAC_BASE + 0x24)))
+#define DMAC_DAR3     (*((vuint32 *)(DMAC_BASE + 0x34)))
+
+/** @} */
+
+/** \defgroup dma_dmatcr_registers DMA Transfer Count Registers (DMATCR0-DMATCR3)
+
+    These registers define the transfer count for each DMA channel. The count
+    is defined as: num_bytes_to_transfer/32
+
+    @{
+ */
+
+#define DMAC_DMATCR0  (*((vuint32 *)(DMAC_BASE + 0x08)))
+#define DMAC_DMATCR1  (*((vuint32 *)(DMAC_BASE + 0x18)))
+#define DMAC_DMATCR2  (*((vuint32 *)(DMAC_BASE + 0x28)))
+#define DMAC_DMATCR3  (*((vuint32 *)(DMAC_BASE + 0x38)))
+
+/** @} */
+
+/** \defgroup dma_chcr_registers DMA Channel Control Registers (CHCR0-CHCR3)
+  
+    These registers configure the operating mode and transfer methodology for 
+    each channel.
+
+    For DMAC_CHCR2, it should always be set to 0x12c1 (source address 
+    incremented, burst mode, interrupt disable, DMA enable).
+
+    For DMAC_CHCR1 and DMAC_CHCR3, it would probably be set to 0x1241
+    (source address incremented, cycle steal mode, interrupt disable,
+    DMA enable).
+
+    @{
+ */
+
+#define DMAC_CHCR0    (*((vuint32 *)(DMAC_BASE + 0x0c)))
+#define DMAC_CHCR1    (*((vuint32 *)(DMAC_BASE + 0x1c)))
+#define DMAC_CHCR2    (*((vuint32 *)(DMAC_BASE + 0x2c)))
+#define DMAC_CHCR3    (*((vuint32 *)(DMAC_BASE + 0x3c)))
+
+/** @} */
+
+
+/** 
+    \brief A register that dictates the overall operation of the DMAC.
+
+    So far we only use it check the status of DMA operations.
+
+    \see    dmaor_operations
```

Comment:
```suggestion
```
Have to remove this since we are making everything inner "anonymous" groups now.